src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1094,21 +1082,19 @@ impl<T: 'static> EventProcessor<T> {
                         if scancode < 0 {
                             return;
                         }
-                        let keysym = wt.xconn.keycode_to_keysym(keycode as ffi::KeyCode);
-                        let virtual_keycode = events::keysym_to_element(keysym as c_uint);
+                        let physical_key = keymap::rawkey_to_keycode(scancode as u32);
                         let modifiers = self.device_mod_state.modifiers();
 
-                        #[allow(deprecated)]
                         callback(Event::DeviceEvent {
                             device_id,
-                            event: DeviceEvent::Key(KeyboardInput {
-                                scancode: scancode as u32,
-                                virtual_keycode,
+                            event: DeviceEvent::Key(RawKeyEvent {
+                                physical_key,
                                 state,
-                                modifiers,
                             }),
                         });
 
+                        // `ModifiersChanged` is dispatched here because we assume that every `XI_KeyPress`
+                        // is preceeded by a `XI_RawKeyPress`.
```

Comment:
It is out of scope of this PR, but I'm not sure this is the right way. `XI_RawKeyPress` is dispatched even if the widow is inactive state because this event is something global.
There is an discussion to provide a way to disable these device events here: https://github.com/rust-windowing/winit/issues/1634
So, in future, this implementation might be problematic.

Reply:
> but I'm not sure this is the right way

I'm not sure of it either, but this is the way it was when I dived into the existing code. The comment is mostly me making assumptions.

Reply:
In any case, there's nothing stopping us from disabling the event when the window is inactive, and re-enabling it when it becomes activated again.